Readers/src/EventInfoReader.cc

```diff
@@ -18,12 +19,15 @@ EventInfoReader::EventInfoReader(const edm::ParameterSet & cfg)
   , analysisConfig_(new AnalysisConfig("produceNtuple", cfg))
   , info_(new EventInfo(*analysisConfig_))
   , runLumiEventReader_(new RunLumiEventReader(cfg))
+  , era_(cfg.getParameter<std::string>("era"))
   , branchName_genHiggsDecayMode_("genHiggsDecayMode")
   , branchName_genWeight_("genWeight")
   , branchName_LHEReweightingWeight_("LHEReweightingWeight")
   , branchName_nLHEReweightingWeight_(Form("n%s", branchName_LHEReweightingWeight_.data()))
   , branchName_htxs_pt_("HTXS_Higgs_pt")
   , branchName_htxs_y_("HTXS_Higgs_y")
+  , branchName_Pileup_nTrueInt_("Pileup_nTrueInt")
+  , pileupCorrectionSet_(LocalFileInPath(Form("TallinnNtupleProducer/EvtWeightTools/data/correctionlib/pu/%s/puWeights.json.gz", era_.data())).fullPath())
```

Comment:
I was thinking that maybe it's better to move this hardcoded piece of information (ie the path name) to here:
https://github.com/HEP-KBFI/TallinnNtupleProducer/blob/bc57d514e1fd6ab59b734431fef452140dac0ed3/Framework/test/produceNtuple_cfg.py#L41
And then read the path straight from the cfg. On the other hand, it should be also done for the CorrectionSet name and I can see how it's going to blow up the number of parameters one needs to specify in the cfg if we want to use `correctionlib` for other purposes than applying PU weights.

If the plan is to keep a copy of the json files here at all times by overwriting them as needed, then we can leave it as it is right now. In that case, @saswatinandan, please add a comment to this PR (not in the code but as a message) where you got these files (URL) and what's their use case (for correcting Run 2 ultralegacy MC?). When overwriting the data files, then the commit message or the PR should state where these files are coming from, so that the next person in the group (or even your future self) doesn't have to figure this out if needed.